lib/beaker/shared/subcommands_util.rb

```diff
@@ -1,22 +1,30 @@
+require 'rake'
+
 module Beaker
   module Shared
     # Methods used in execution of Subcommands
-    # - should we write the config?
-    # - reset ARGV 
+    # - should we execute a subcommand?
+    # - reset ARGV
     # - execute Beaker
+    # - update a rakefile to require beaker quickstart tasks
+    # - execute a rake task
+    # - execute the vagrant quickstart task
+    # - execute the vmpooler quickstart task
+    # - exit with a specific message
+    # - verify that the init subcommand has the correct arguments
+    # - execute the quick start task for the specified hypervisor
     module SubcommandsUtil
-      CONFIG_PATH = ".beaker/config"
-    
-      @@write_config = false
+      RAKE_FILES = ["Rakefile", "rakefile", "Rakefile.rb", "rakefile.rb"]
+      BEAKER_REQUIRE = "require 'beaker/tasks/quick_start'"
+      HYPERVISORS = ["vagrant", "vmpooler"]
 
-      def self.write_config=( val )
-        @@write_config = val
-      end
-      
-      def self.write_config?
-        @@write_config
+      # Check if the first argument to the beaker execution is a subcommand
+      # @return [Boolean] true if argv[0] is "help" or a method defined in the Subcommands class, false otherwise
+      def execute_subcommand?(arg0)
+        return false if arg0.nil?
+        (Beaker::Subcommands.instance_methods(false) << :help).include? arg0.to_sym
```

Comment:
Yep! `beaker --help` will execute old school help, `beaker help` will execute the new subcommand help